### PR TITLE
Style Select

### PR DESCRIPTION
**Pull request by jakubsy**

### CODE REVIEW

File: packages/neptune-extension/src/common/components/checkout-modal/CheckoutModal.less [hunk 1]
```diff
@@ -0,0 +1,5 @@
+.checkout-modal {
+  &__input {
```
Comment (themithy) [inline packages/neptune-extension/src/common/components/checkout-modal/CheckoutModal.less]: Would you mind refactoring this to one `Input` component ? We already have such quickfix code in every modal that uses input.

File: packages/neptune-extension/src/common/utils/projects.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { backendClient } from '../api/backend-client';
+
+export async function fetchProjects() {
```
Comment (themithy) [inline packages/neptune-extension/src/common/utils/projects.js]: Would you mind looking at `common/utils/checkout.ts`, it's already implemented there.

File: packages/neptune-extension/src/common/components/upload-modal/UploadModal.tsx [hunk 3]
```diff
@@ -78,14 +81,19 @@ const UploadModal: React.FC<UploadModalProps> = ({
...
-      : window.localStorage.getItem(PROJECT_LOCAL_STORAGE_KEY) || ''
-  });
+  const defaultProjectId = !!notebook
+    ? notebook.projectId
+    : window.localStorage.getItem(PROJECT_LOCAL_STORAGE_KEY) || '';
+
   const [ name, setName ] = React.useState('');
   const [ description, setDescription ] = React.useState('');
 
+  const [ projectId, projectInputProps, projectMetaProps, setProjectId ] = useSelectInputValue(
```
Comment (themithy) [inline packages/neptune-extension/src/common/components/upload-modal/UploadModal.tsx]: Here you should use `projectInputProps.onChange` and not change the interface.

File: packages/neptune-extension/src/common/components/input/SelectInput.tsx [hunk 4]
```diff
@@ -1,34 +1,29 @@
...
   options,
   onChange,
+  ...rest
 }) => {
 
   return (
-    <React.Fragment>
-      {label}
-      <select
+    <ValidationWrapper>
```
Comment (themithy) [inline packages/neptune-extension/src/common/components/input/SelectInput.tsx]: Here you embeded the `ValidationWrapper` inside input component, and in other places (e.g. UploadModal) it is inlined with the input code. I guess we should work towards establishing one solution or the other.

File: packages/neptune-extension/src/common/hooks/useSelectInputValue.ts [hunk 5]
```diff
@@ -9,15 +9,14 @@ export interface SelectInputProps {
 }
 
 export interface SelectMetaProps {
-  valid?: boolean
-  loading?: boolean
+  status: 'pending' | 'success' | 'error'
```
Comment (themithy) [inline packages/neptune-extension/src/common/hooks/useSelectInputValue.ts]: This code is meant to be two flags, one representing the promise (loading), and the other representing the validity (valid). You rewrote it to one promise flag, which is something different logically. It is difficult to see the reason behind this change, because the old interface supports the usecase you were going to implement. This PR says "Style Select", but you rewrote a lot of unrelated code.

File: packages/neptune-extension/src/common/hooks/useSelectInputValue.ts [hunk 6]
```diff
@@ -59,12 +58,10 @@ export default function useSelectInputValue(
     options,
     onChange,
   };
-
   const metaProps: SelectMetaProps = {
-    valid,
-    loading,
+    status: loading ? 'pending' : valid ? 'success' : 'error',
```
Comment (themithy) [inline packages/neptune-extension/src/common/hooks/useSelectInputValue.ts]: Would you mind working on this, because I guess such use of the ternary operator is discouraged by most js guidelines.

File: packages/neptune-extension/src/common/components/checkout-modal/CheckoutModal.tsx [hunk 7]
```diff
@@ -77,46 +84,61 @@ const CheckoutModal: React.FC<CheckoutModalProps> = ({
     onClose();
   }
 
-  const disabled = !projectMetaProps.valid || !notebookMetaProps.valid || !checkpointMetaProps.valid;
+  const disabled = projectMetaProps.status === 'error' || notebookMetaProps.status === 'error' || checkpointMetaProps.status === 'error';
```
Comment (themithy) [inline packages/neptune-extension/src/common/components/checkout-modal/CheckoutModal.tsx]: This one is wrong, the form is meant to be disabled when any value in invalid, not when it is invalid **and** finished loading.

File: packages/neptune-extension/src/common/components/checkout-modal/CheckoutModal.less (hunk 1 above)
Comment (jakubsy) [reply packages/neptune-extension/src/common/components/checkout-modal/CheckoutModal.less]: I dont think Input by default should have any flex properties.

File: packages/neptune-extension/src/common/components/checkout-modal/CheckoutModal.tsx (hunk 7 above)
Comment (jakubsy) [reply packages/neptune-extension/src/common/components/checkout-modal/CheckoutModal.tsx]: Ok, I'll fix that

File: packages/neptune-extension/src/common/components/upload-modal/UploadModal.tsx (hunk 3 above)
Comment (jakubsy) [reply packages/neptune-extension/src/common/components/upload-modal/UploadModal.tsx]: Ok

File: packages/neptune-extension/src/common/utils/projects.js (hunk 2 above)
Comment (jakubsy) [reply packages/neptune-extension/src/common/utils/projects.js]: Ok

File: packages/neptune-extension/src/common/components/input/SelectInput.tsx (hunk 4 above)
Comment (jakubsy) [reply packages/neptune-extension/src/common/components/input/SelectInput.tsx]: There is simple input and simple select.
I reused the SelectInput to avoid code repetition and I would like to not change that.

File: packages/neptune-extension/src/common/components/checkout-modal/CheckoutModal.less (hunk 1 above)
Comment (themithy) [reply packages/neptune-extension/src/common/components/checkout-modal/CheckoutModal.less]: This depends, if the `flex` props is needed to display the input properly, than definitely it should be moved there. But if it's needed only in some case, then this case should be factored out.

What is the reason to use `flex` here ?

Comment (jakubsy) [reply packages/neptune-extension/src/common/components/checkout-modal/CheckoutModal.less]: Flex is needed to be displayed properly only in selected cases (flex-form).
I can imagine using Input outside of flex container and there flex is not needed
